syntax = "proto3";
package images;

message Image {
  int32 width = 1;
  int32 height = 2;
  string id = 3;
  bytes imgdata = 4;
}

message ObjectDetection {
    int32 left = 1;
    int32 right = 2;
    int32 top = 3;
    int32 bottom = 4;
    float score = 5;
    string class_name = 6;
}

message ImageObjectDetections {
    Image image = 1;
    repeated ObjectDetection detections = 2;
    string algorithm_name = 3;
}
